react/ConditionContext.ts

```diff
@@ -18,19 +20,34 @@ export const ConditionContext = createContext<ConditionContextValue>(
 
 export const ConditionDispatchContext = createContext<Dispatch<Actions>>(Noop)
 
-export function reducer(prevState: ConditionContextValue, action: Actions) {
-  if (action.type === 'UPDATE_MATCH') {
-    if (prevState.matched === action.payload.matches) {
-      return prevState
+export function reducer(
+  prevState: ConditionContextValue,
+  action: Actions
+): ConditionContextValue {
+  switch (action.type) {
+    case 'SET_VALUES': {
+      return {
+        ...prevState,
+        values: action.payload.values,
+      }
     }
 
-    return {
-      ...prevState,
-      matched: Boolean(prevState.matched) || action.payload.matches,
+    case 'UPDATE_MATCH': {
+      const { matches } = action.payload
+
+      if (prevState.matched === matches || matches == null) {
+        return prevState
+      }
+
+      return {
+        ...prevState,
+        matched: Boolean(prevState.matched) || matches,
+      }
```

Comment:
I don't know exactly what are the requirements for this app and I didn't find any explanation to this (maybe I just didn't look at all the documentation), but why I can't update the `matched` value from `true` to `false`?

Reply:
We can't override the value If some condition resolves to `false` but another one has already resolved to `true`:

`matched: true || false` vs `matched: false`